server/src/main/java/org/apache/druid/catalog/model/table/HttpTableDefn.java

```diff
@@ -0,0 +1,190 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.catalog.model.table;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import org.apache.druid.catalog.model.CatalogUtils;
+import org.apache.druid.catalog.model.ModelProperties.StringListPropertyDefn;
+import org.apache.druid.catalog.model.ModelProperties.StringPropertyDefn;
+import org.apache.druid.catalog.model.ParameterizedDefn;
+import org.apache.druid.catalog.model.ResolvedTable;
+import org.apache.druid.catalog.model.table.ExternalTableDefn.FormattedExternalTableDefn;
+import org.apache.druid.data.input.InputSource;
+import org.apache.druid.data.input.impl.HttpInputSource;
+import org.apache.druid.java.util.common.IAE;
+import org.apache.druid.java.util.common.ISE;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.metadata.DefaultPasswordProvider;
+import org.apache.druid.metadata.EnvironmentVariablePasswordProvider;
+import org.apache.druid.utils.CollectionUtils;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Definition of an input table for an HTTP data source. Provides the same
+ * properties as the {@link HttpInputSource}, but as top-level properties
+ * that can be mapped to SQL function parameters. Property names are
+ * cleaned up for ease-of-use. The HTTP input source has multiple quirks,
+ * the conversion method smooths over those quirks for a simpler catalog
+ * experience. Provides a parameterized
+ * form where the user provides the partial URLs to use for a particular
+ * query.
+ */
+public class HttpTableDefn extends FormattedExternalTableDefn implements ParameterizedDefn
+{
+  public static final String TABLE_TYPE = HttpInputSource.TYPE_KEY;
+  public static final String URI_TEMPLATE_PROPERTY = "template";
+  public static final String USER_PROPERTY = "user";
+  public static final String PASSWORD_PROPERTY = "password";
+  public static final String PASSWORD_ENV_VAR_PROPERTY = "passwordEnvVar";
+  public static final String URIS_PROPERTY = "uris";
+  public static final String URIS_PARAMETER = "uris";
```

Comment:
why are URIS both a property and parameter?

Reply:
Because it is? I can specify it _either_ in the catalog (property) or in a SQL table statement (parameter). Not all properties are parameters and visa-versa. In this case, `uris` is both.